common/beerocks/bwl/dummy/sta_wlan_hal_dummy.cpp

```diff
@@ -17,7 +17,8 @@ namespace bwl {
 namespace dummy {
 
 sta_wlan_hal_dummy::sta_wlan_hal_dummy(const std::string &iface_name, hal_event_cb_t callback)
-    : base_wlan_hal(), base_wlan_hal_dummy(bwl::HALType::Station, iface_name, callback, {})
+    : base_wlan_hal(bwl::HALType::Station, iface_name, IfaceType::Intel, callback, {}),
```

Comment:
If base_wlan_hal cannot be safely default-constructed, maybe we could delete the default constructor?
It would require every derived class to call it explicitly so it's more work, but it would avoid the same issue from happening again in the future (I didn't go through the full inheritance tree, so I might have missed something).